core/unix/loader_linux.c

```diff
@@ -454,42 +487,47 @@ privload_tls_init(void *app_tp)
     /* GDB reads some pthread members (e.g., pid, tid), so we must make sure
      * the size and member locations match to avoid gdb crash.
      */
-    ASSERT(TLS_PRE_TCB_SIZE == LIBC_PTHREAD_SIZE);
+    ASSERT(TLS_PRE_TCB_SIZE >= LIBC_PTHREAD_SIZE);
     ASSERT(LIBC_PTHREAD_TID_OFFSET == offsetof(dr_pthread_t, tid));
 #endif
     LOG(GLOBAL, LOG_LOADER, 2, "%s: allocated %d at " PFX "\n", __FUNCTION__,
         client_tls_alloc_size, dr_tp);
 #ifdef RISCV64
```

Comment:
Can do ifndef X86 instead of duplicating for AARCHXX.

Reply:
Done.